docs/sources/developers/plugins/backend/debug-your-backend-plugin.md

```diff
@@ -0,0 +1,69 @@
++++
+title = "Debug your backend plugin"
+keywords = ["grafana", "plugins", "backend", "plugin", "backend-plugins", "documentation"]
+type = "docs"
++++
+
+# Debug your backend plugin
+
+This guide explains how to attach a debugger to a running backend plugin.
+
+A backend plugin runs in its own process, separate from grafana-server. This makes it difficult to debug the plugin the same way you would debug the grafana-server process.
+
+You can read more about the how plugin processes are managed in the [go-plugin](https://github.com/hashicorp/go-plugin#architecture) project page.
+
+Fortunately, you can debug your plugin by attaching the [Delve](https://github.com/go-delve/delve) debugger for Go to the subprocess running your plugin.
+
+You can add more information available while debugging by building your plugin with additional flags.
+
+1. Disable compiler optimizations and inlining by adding `-gcflags=all="-N -l"` when you build your plugin.
+
+   ```
+   go build -gcflags=all="-N -l" -o ./dist/my-datasource_linux_amd64 ./backend
```

Comment:
I tried to get this running with the `grafana/azure-data-explorer-datasource` but did not manage to set it up according to the guide.

Some suggestions to improve it would be to add a debug build schema to the: https://github.com/grafana/grafana-plugin-sdk-go/blob/master/build/common.go

Any how I would love to jump on a call to get this working so I can give better feedback on how to make this guide a bit easier (foolproof) to follow.

Reply:
I think @alexanderzobnin would be the best one suited for this. I'd love to join to understand how to improve the guide.